app/models/king.rb

```diff
@@ -1,2 +1,16 @@
 class King < Piece
+  def valid_move?(_destination_row, _destination_col)
+    kings_direction = [
+      [1, 0],
+      [-1, 0],
+      [0, -1],
+      [0, 1],
+      [1, -1],
+      [1, 1],
+      [-1, -1],
+      [-1, 1]
+    ]
+    return true if kings_direction.include?(direction)
```

Comment:
Where does `direction` come from?

You'll want this method to be using `destination_row` &  `destination_col`.

Because we'll be calling `valid_move?` before we move the piece. To be sure the move is OK.